docs/wiki/The CI environment - Pipeline design.md

```diff
@@ -0,0 +1,146 @@
+This section provides an overview of the design principles followed by the CARML CI environment pipelines.
+
+---
+
+### _Navigation_
+
+- [Module Pipelines](#module-pipelines)
+  - [Pipeline phases](#pipeline-phases)
+    - [DevOps-Tool-specific design](#devops-tool-specific-design)
+  - [Module pipeline inputs](#module-pipeline-inputs)
+- [Platform pipelines](#platform-pipelines)
+  - [Dependencies pipeline](#dependencies-pipeline)
+    - [Dependencies pipeline inputs](#dependencies-pipeline-inputs)
+  - [ReadMe pipeline](#readme-pipeline)
+  - [Wiki pipeline](#wiki-pipeline)
+
+---
+
+# Module Pipelines
+
+The repository hosts one pipeline for each module in the CARML library.
+
+The purpose of each module pipeline is twofold:
+
+1. **Validation**: To ensure the modules hosted by the CARML library are valid and can perform the intended deployments.
+1. **Publishing**: To publish _versioned_ and already validated modules to one or multiple target locations, from where they can be referenced by solutions consuming them.
+
+As such each pipeline can be mapped to `Phases 1 and 2` described in the [Deployment flow](./The%20context%20-%20CARML%20CI%20environment#deployment-flow) section.
+
+<img src=".\media\CIEnvironment\pipelineDesign.png" alt="Pipeline phases" height="500">
+
+The following paragraphs provide an overview of the different phases and shared logic the module pipelines use.
+
+- [Pipeline phases](#pipeline-phases)
+- [Module pipeline inputs](#module-pipeline-inputs)
+
+## Pipeline phases
+
+This paragraph provides an overview of the three phases executed by each module pipeline. Further details about the implementation and design of each phase are provided in the dedicated pages linked below.
+
+1. **Static Validation**: Executes a set of static Pester tests against the module template, to ensure they comply with the intended module design principles. Further details for this phase are provided by the corresponding Wiki section [Static validation](./The%20CI%20environment%20-%20Static%20validation).
```

Comment:
For some reason I cannot add a suggestion and have to just add a comment:

1. **Static Validation**: Executes a set of static Pester tests against the module and its templates to ensure they comply with our design principles. Further details for this phase are provided by the corresponding Wiki section [Static validation](./The%20CI%20environment%20-%20Static%20validation).

Reply:
```suggestion
1. **Static Validation**: Executes a set of static Pester tests against the module and its templates to ensure they comply with our design principles. Further details for this phase are provided by the corresponding Wiki section [Static validation](./The%20CI%20environment%20-%20Static%20validation).
```